BUILDING.md

```diff
@@ -0,0 +1,276 @@
+`install` subdirectory of the working directory, so if you have been
+following the instructions above and the build worked, you should see
+an installation tree under `build/install`.
+
+If the build did not succeed, please see the next section.
+
+
+## Options to Configure or Customize the Build
+
+### Options for Package Search
+
+A frequent cause of difficulty or failure is an inability for the
+build to find a third-party dependency like `gRPC`. If the build
+fails, first check if you have installed the right dependencies per
+the lists above. In some cases, you may find that a necessary direct
+or transitive dependency has been installed, but to a location that
+CMake does not search by default.
+
+#### Setting `PKG_CONFIG_PATH`
+
+On Homebrew, OpenSSL does not get installed in the default search
+paths because it is "keg-only". We can inform CMake of the actual
+install location by customizing `PKG_CONFIG_PATH` in the shell
+environment before invoking CMake:
+
+``` shell
+PKG_CONFIG_PATH="/opt/homebrew/opt/openssl@3/lib/pkgconfig" cmake ...
+```
+
+#### Setting `CMAKE_PREFIX_PATH`
+
+CMake also has its own intrinsic package search mechanism. If you have
+dependencies installed to locations that are not part of CMake's
+default search set, you can add them by configuring
+`CMAKE_PREFIX_PATH` as an argument to CMake. For instance, if you have
+a personal tree of installed software in `$HOME/opt`, you can inform
+CMake that this path should be searched for packages by passing it as
+a command line "define" to CMake:
+
+``` shell
+cmake -DCMAKE_PREFIX_PATH=$HOME/opt ...
+
+```
+
+Please see
+[CMAKE_PREFIX_PATH](https://cmake.org/cmake/help/latest/variable/CMAKE_PREFIX_PATH.html)
+for more details on `CMAKE_PREFIX_PATH`.
+
+### Options for Building the SDK
+
+### `CMAKE_INSTALL_PREFIX`
+
+By default, the SDK will install into a subdirectory of the working
+build directory. This is not appropriate for making the SDK available
+for other projects. To install the SDK to a location of your choice
+you should customize `CMAKE_INSTALL_PREFIX` while invoking `CMake` and
+then re-run the `install` task. For instance, to install the SDK to
+`/usr/local`:
+
+
+``` shell
+cmake -DCMAKE_INSTALL_PREFIX=/usr/local ...
+ninja install
+```
+
+#### `VIAMCPPSDK_ENFORCE_COMPILER_MINIMA`
+
+By default, the SDK build will enforce compiler minima. If you must
+bypass this check for some reason, you may do so by disabling this
+default-on option:
+
+``` shell
+cmake -DVIAMCPPSDK_ENFORCE_COMPILER_MINIMA=OFF ...
+```
+
+#### `VIAMCPPSDK_USE_DYNAMIC_PROTOS`
+
+By default, the SDK build will use checked-in copies of generated
+sources. However, these sources only work with a narrow range of gRPC
+versions. If the checked-in copies do not work, or if you would prefer
+to always work against regenerated code, you can request that proto
+generation happen as part of the build by enabling this flag:
+
+``` shell
+cmake -DVIAMCPPSDK_USE_DYNAMIC_PROTOS=ON ...
```

Comment:
Should this be the default? What's the situation where you don't want to generate these fresh? If you're not modifying the API (which most external users shouldn't be) it seems like you'd want this almost always.

Reply:
Yes, there was a missing dependency edge, which I believe I've fixed in the most recent push. I thought I'd explained to CMake what I was doing, but I in fact had not.

Regarding whether `DYNAMIC_PROTOS` should be the default, I can't really speak to that. The project contains pre-generated files checked in, so I went with using those by default. Otherwise, why are they there?

I'm happy to decide that they shouldn't be there and that we should always generate, but I'd prefer to do that after merging what I've got. I'd like a call from @stuqdog and @njooma about how they want it to work. I don't really have enough insight to have an opinion.

Reply:
I'm fine with it not being default, and agree that can be something for a later PR, but as it appears to be absolutely required under Debian (at least) can you at least make loud note about it up near the top (maybe where you mention troubleshooting?)

Reply:
Added the following:
```diff
diff --git a/BUILDING.md b/BUILDING.md
index 35e440c..d3cef86 100644
--- a/BUILDING.md
+++ b/BUILDING.md
@@ -16,6 +16,12 @@ If you experience problems while following this guide, please see the
 `Limitations, Known Issues, and Troubleshooting` section at the bottom
 to see if your issue (and hopefully a workaround) is covered there.
 
+PLEASE NOTE: It is very likely that you will need to rebuild the
+generated sources as part of building the SDK. Please see the
+documentation on `VIAMCPPSDK_USE_DYNAMIC_PROTOS` below, and the
+additional comments about mismatched protobufs in the troubleshooting
+guide, before continuing further.
+
 ## Software Prerequisites
 
 The project depends on [CMake](https://cmake.org/) >= 3.25,
```